### ISSUE TITLE
How to switch CentOS to Oracle Linux with local yum repository (NO ISP).

### ISSUE THREAD

**Issue by petrjonak1** [oracle/centos2ol#83]

**Comment by petrjonak1** [comment]
Some idea how to do it ?

**Comment by Djelibeybi** [comment]
At a high level and assuming you're migrating from CentOS 8 to Oracle Linux 8:,

- [Set `yum_url` to the FQDN of the your mirror server](https://github.com/oracle/centos2ol/blob/main/centos2ol.sh#L12)
- [Change the filename of the `repo_file` to one that has your mirror details](https://github.com/oracle/centos2ol/blob/main/centos2ol.sh#L142)
- Change the URLs for [`ol8_baseos_latest`](https://github.com/oracle/centos2ol/blob/main/centos2ol.sh#L294), [`ol8_appstream`](https://github.com/oracle/centos2ol/blob/main/centos2ol.sh#L301) and [`ol8_UEKR6`](https://github.com/oracle/centos2ol/blob/main/centos2ol.sh#L308) to your mirror

You then need to add some logic [after the initial switch is finished](https://github.com/oracle/centos2ol/blob/main/centos2ol.sh#L489) to fix all the repo URLs again because the ones you set above will be replaced by the switch process. You need to do this before the [script calls `yum distro-sync`](https://github.com/oracle/centos2ol/blob/main/centos2ol.sh#L491).

**Comment by klokanovic** [comment]
Hi,

I am trying to convert a C7 server to OEL7 using my local repository.

Even with yum_url set to my mirror and repo_file set with my local details - the script still looks for the internet repos at this point:

+ echo 'Installing base packages for Oracle Linux...Installing base packages for Oracle Linux...
+ yum shell -y
https://yum.oracle.com/repo/OracleLinux/OL7/UEKR6/x86_64/repodata/repomd.xml: [Errno 14] curl#6 - "Could not resolve host: yum.oracle.com; Name or service not known"
Trying other mirror.

 One of the configured repositories failed (Latest Unbreakable Enterprise Kernel Release 6 for Oracle Linux 7Server (x86_64)),

I can't at all see where I need to 'fix all the repo URLs again because the ones you set above will be replaced by the switch process' - exactly where does it make the switch?

Thanks to any responses.

**Comment by Djelibeybi** [comment]
It would probably be easier just to use your local DNS server (or `/etc/hosts` file if that's not possible) to resolve `yum.oracle.com` to your local mirror instead, to be honest.

**Comment by talsh5528** [comment]
how should I know which packages i should add to my local repo? in order to support all the packages that need to be installed in the migration process? @Djelibeybi

**Comment by Djelibeybi** [comment]
The simplest solution is just to mirror all the packages using either `uln-yum-mirror` (available in the `ol7_addons` or `ol8_addons` repo) or my container-based solution which can run on any host that has Docker or Podman installed (including Windows or macOS using Docker Desktop): https://github.com/djelibeybi/oraclelinux-reposync

**Comment by talsh5528** [comment]
thanks for the fast answer, and if I want to continue the process of line and migrate from Oracle 7 to Oracle 8 using Leapp command, there is also a nice solution?

From: Avi Miller ***@***.***>
Date: Sunday, 23 July 2023 at 17:14
To: oracle/centos2ol ***@***.***>
Cc: Tal Shechanovitz ***@***.***>, Comment ***@***.***>
Subject: Re: [oracle/centos2ol] How to switch CentOS to Oracle Linux with local yum repository (NO ISP). (#83)

The simplest solution is just to mirror all the packages using either uln-yum-mirror (available in the ol7_addons or ol8_addons repo) or my container-based solution which can run on any host that has Docker or Podman installed (including Windows or macOS using Docker Desktop): https://github.com/djelibeybi/oraclelinux-reposync

**Comment by Djelibeybi** [comment]
You would need complete mirrors of both the Oracle Linux 7 and Oracle Linux 8 repos, as documented: https://docs.oracle.com/en/operating-systems/oracle-linux/8/leapp/leapp-AboutLeapp.html#upgrade-cond

**Comment by Djelibeybi** [comment]
For Leapp to work, you need to mirror the entire repository, otherwise it's liable to fail to upgrade and you'll be left with broken instances. To find the current size of the repos, run `yum repolist -v` on the OL8 machine. Looks like `ol8_baseos_latest` is 46GB and `ol8_appstream` is 374GB.

**Comment by talsh5528** [comment]
so there is another way to migrate the oracle7 machine to oracle8 in offline env? without saving 400GB +? 
more I tried to migrate with my local repo after mirroring only the newest version of each package, but the leapp command failed:
`1. Inhibitor: Some required custom target repositories have not been found
Consult the pre-upgrade report for details and possible remediation.`
@Djelibeybi

**Comment by Djelibeybi** [comment]
No, there isn't. Switching from CentOS 7 to Oracle Linux 7 then upgrading to Oracle Linux 8 is very, very tricky. You're probably better off just installing a new Oracle Linux 8 instance offline and moving your application manually.

**Comment by talsh5528** [comment]
> Djelibeybi
i have jest finished cloning all packages from ol8_baseos_latest and ol8_appstream(the newest version), and I have my local repo, but it looks like the leapp upgrade don't recognize my repo, even when I am using this command
`sudo leapp preupgrade --oraclelinux --enablerepo=ol8_baseos_local` 
I get the error I mentioned earlier:
![image](https://github.com/oracle/centos2ol/assets/118183238/2ec72502-68ec-44d3-a0ce-abb7f310864d)
there is some way to handle that? 
ן עק

**Comment by Djelibeybi** [comment]
This isn't the right place for Leapp support. I suggest you review the Upgrade documentation very carefully and check the logs to determine what's missing. The preflight checks should guide you to the right answer. Otherwise, you might need to use Oracle Support.

**Comment by hushamyaqoob** [comment]
> It would probably be easier just to use your local DNS server (or `/etc/hosts` file if that's not possible) to resolve `yum.oracle.com` to your local mirror instead, to be honest.


I have a local repo server with HTTP but the genrated URL goes to HTTPS. How and where to change it. i cant find a please to change https to http


https://192.168.88.35/repo/OracleLinux/OL7/UEKR6/x86_64/repodata/repomd.xml: [Errno 14] curl#6 - "Could not resolve host: yum.oracle.com; Name or service not known"
Trying other mirror.

**Comment by scoter-oracle** [comment]
Script has been revamped. Please re-test and re-open the issue if it still fails.
